server/src/main/java/io/crate/execution/ddl/tables/DropColumnTask.java

```diff
@@ -96,13 +100,42 @@ public ClusterState execute(ClusterState currentState, DropColumnRequest request
                 }
             }
         }
+        // Add relevant constraints to be removed
+        Set<String> constraintsToRemove = new HashSet<>();
+        for (var constraint : currentTable.checkConstraints()) {
+            Set<ColumnIdent> columnsInConstraint = new HashSet<>();
+            RefVisitor.visitRefs(constraint.expression(), r -> columnsInConstraint.add(r.column()));
+
+            if (columnsInConstraint.size() == 1) {
+                for (var refToDrop : refsToDrop) {
+                    if (columnsInConstraint.contains(refToDrop.column())) {
+                        constraintsToRemove.add(constraint.name());
+                    }
+                }
+            }
+            if (columnsInConstraint.stream().allMatch(
+                columnInConstraint -> {
+                    for (var refToDrop : refsToDrop) {
+                        if (columnInConstraint.isChildOf(refToDrop.column())) {
+                            return true;
+                        }
+                    }
+                    return false;
+                })) {
+                constraintsToRemove.add(constraint.name());
+            }
```

Comment:
Couldn't this be simplified?
```suggestion
            if (columnsInConstraint.stream().allMatch(
                columnInConstraint -> {
                    for (var refToDrop : refsToDrop) {
                        if (columnInConstraint.equals(refToDrop.column()) || columnInConstraint.isChildOf(refToDrop.column())) {
                            return true;
                        }
                    }
                    return false;
                })) {
                constraintsToRemove.add(constraint.name());
            }
```

Reply:
Yep, thx, I thought wrong that we still need the size() == 1 check, but it's already checked again, the analyzer validation is checked again here based on the most recent clusterState.